spec/uri_parser_spec.rb

```diff
@@ -56,4 +56,19 @@ def self.describe_parsed(url, &block)
     its(:uri) { should == 'http://domain.com/?b=&c&d=5' }
     its(:query_params) { should == { 'b' => nil, 'c' => nil, 'd' => '5' } }
   end
+
+  describe_parsed 'http://domain.com/?&c=3&d=5' do
+    its(:uri) { should == 'http://domain.com/?&c=3&d=5' }
+    its(:query_params) { should == { nil => nil, 'c' => '3', 'd' => '5' } }
+  end
+
+  describe_parsed 'http://domain.com/?c=3&&d=5' do
+    its(:uri) { should == 'http://domain.com/?c=3&&d=5' }
+    its(:query_params) { should == { 'c' => '3', nil => nil, 'd' => '5' } }
```

Comment:
Given that:
```
its(:uri) { should == 'http://domain.com/?c=3&d=5&' }
its(:query_params) { should == { 'c' => '3', 'd' => '5' } }
```
I'd expect that the `nil => nil` wouldn't be in the output, since both urls `http://domain.com/?c=3&d=5&` and `http://domain.com/?c=3&&d=5` represent the same thing.  

Is there a scenario where these urls are effectively different, and we'd want a `nil => nil` in one but not the other?

If not, could we compact the results to remove the `nil => nil`?

Reply:
I agree, I think `nil => nil` shouldn't be in the output. 

I have found an example of a real url with `&&` in the query string, where replacing `&&` with `&` actually does point to a different page. I don't really know how to best represent that in this uri parser.

Reply:
👍 , especially if the output is a hash (`{'c' => '3', 'd' => '5'}`) which are unordered it'll be really hard to reflect ordering.

How frequent is the example where `&&` and `&` point to a different page?  If it happens rarely, I'm not sure if it matters a ton and it'd be OK w/o worrying about ordering.

Reply:
It doesn't happen often, from what I've seen.